dashboard/src/utils/dateFormatter.js

```diff
@@ -1,3 +1,9 @@
 export const formatDate = (date) => {
-  return `${date.getFullYear()}-0${date.getMonth() + 1}-${date.getDate()}`;
+  const utcdateMonth =
+    date.getUTCMonth() + 1 < 10
+      ? `0${date.getUTCMonth() + 1}`
+      : date.getUTCMonth() + 1;
+  const utcDate =
+    date.getUTCDate() < 10 ? `0${date.getUTCDate()}` : date.getUTCDate();
+  return `${date.getUTCFullYear()}-${utcdateMonth}-${utcDate}`;
```

Comment:
_[I am repeatedly amazed by what one can and cannot do in Javascript....]_

Apparently, converting numbers to strings with leading zeros is a non-trivial operation in Javascript.  How about something like this:
```
  const pad = num => num < 10 ? '0'+num : num;
  const mNum = date.getUTCMonth() + 1;
  const dNum = date.getUTCDate();
  return `${date.getUTCFullYear()}-${pad(mNum)}-${pad(dNum)}`;
```
Or, you could consider doing this:
```
  return date.toLocaleDateString('sv', {
    "timeZone": "UTC",
    "year": "numeric",
    "month": "2-digit",
    "day": "2-digit",
  });
```
(Yeah, that's the _Swedish_ locale...because the Internet sez it's the best match for ISO 8601.  😝)  NB:  `timeZone` has a capital `Z` -- Javascript doesn't complain if you use a lower case `z`...it just silently ignores the option and you get the wrong answer some of the time.  😬

Reply:
The first option looks more clean to me.If we want to go for the second approach(which is also quite interesting),we might need to add few comments on what are we trying to achieve with this piece of code(since it's not that trivial at first look).

Reply:
@webbnh,The first code will boil down to something like this (In your code,you are adding padding to the number and then again concatenating the padded number with the number we get from UTC methods,so I am providing my version of your code)

 ```
const mNum=date.getUTCMonth()+1
const dNum=date.getUTCDate()
const padMonth = mNum < 10 ? '0'+mNum : mNum
const padDate=dNum<10?'0'+dNum:dNum
  return `${date.getUTCFullYear()}-${padMonth}-${padDate}`;

Reply:
@shubham-html-css-js, I'm not understanding what you're trying to describe.

In my code, I put the padding code into a callable, so that it could be used for each of the day and the month; you're in-lining the padding code, which is fine, but it duplicates the code and requires an extra line.

(Did my code not work for you?  It worked for me....  😏)

Reply:
Ok,my bad,I had a blind spot moment

Reply:
Can we move these data handling changes to follow-on PR?

Reply:
I think we could, but I also think that Shubham has this working, now.

Reply:
Great.